playerbot/strategy/druid/DruidActions.h

```diff
@@ -110,10 +130,14 @@ namespace ai
 		CastWrathAction(PlayerbotAI* ai) : CastRangeSpellAction(ai, "wrath") {}
 	};
 
-	class CastStarfallAction : public CastRangeSpellAction
+	class CastStarfallAction : public CastBuffSpellAction
 	{
 	public:
-		CastStarfallAction(PlayerbotAI* ai) : CastRangeSpellAction(ai, "starfall") {}
+		CastStarfallAction(PlayerbotAI* ai) : CastBuffSpellAction(ai, "starfall") {}
+		virtual bool isUseful()
+		{
+			return sServerFacade.IsDistanceLessOrEqualThan(AI_VALUE2(float, "distance", GetTargetName()), 25.0f);
+		}

```

Comment:
If IsUseful() is false I guess the caster will not go to the target Position. Check how CastMeleeSpellAction works. There a range is set and because it inherits from RangeSpell the Bot will automatically try to reach that Range - just an Idea

Reply:
i figured out that buffspell already got a distance check - so removing this is ok for me, rihgt?